app/controllers/assignments_controller.rb

```diff
@@ -3,24 +3,24 @@ class AssignmentsController < ApplicationController
   before_filter :admin_only
 
   def knight
-    v = Volunteer.find(params[:volunteer_id])
-    r = Region.find(params[:region_id])
-    unless current_volunteer.any_admin?(r)
+    volunteer = Volunteer.find(params[:volunteer_id])
+    region = Region.find(params[:region_id])
+    unless current_volunteer.any_admin?(region)
       flash[:notice] = "Not permitted to do knightings in that region..."
       redirect_to(root_path)
       return
```

Comment:
Can you change this to return redirect_to ? http://stackoverflow.com/questions/5743534/redirect-to-return